File: lib/src/main/java/arsatech/co/utils/TimberLogTree.java
package arsatech.co.utils;

import android.annotation.SuppressLint;
import android.support.annotation.Nullable;
import android.util.Log;

import timber.log.Timber;

/**
 * Created by Aref Bahreini Nejad on 27/04/2019.
 * Updated on 27/04/2019
 */
public class TimberLogTree extends Timber.Tree {

	private static final String DEFAULT_TAG = "TimberLogTree";

	@SuppressLint("LogNotTimber")
	@Override
	protected void log(int priority, @Nullable String tag, @Nullable String message, @Nullable Throwable t) {
		if (!BuildConfig.DEBUG && (priority == Log.VERBOSE || priority == Log.DEBUG || priority == Log.INFO)) {
			return;
		}
		switch (priority) {
			case Log.VERBOSE:
				if (tag != null)
					Log.v(tag, message);
				else
					Log.v(DEFAULT_TAG, message);
				break;
			case Log.DEBUG:
				if (tag != null)
					Log.d(tag, message);
				else
					Log.d(DEFAULT_TAG, message);
				break;
			case Log.INFO:
				if (tag != null)
					Log.i(tag, message);
				else
					Log.i(DEFAULT_TAG, message);
				break;
			case Log.WARN:
				if (tag != null)
					Log.w(tag, message);
				else
					Log.w(DEFAULT_TAG, message);
				break;
			case Log.ERROR:
				if (tag != null)
					Log.e(tag, message);
				else
					Log.e(DEFAULT_TAG, message);
				break;
		}
	}

}

File: lib/src/main/kotlin/com/arefdev/base/utils/TimberLogTree.kt
package com.arefdev.base.utils

import android.annotation.SuppressLint
import android.util.Log
import timber.log.Timber
import java.util.Locale

/**
 * Updated on 21/09/2023
 *
 * @author <a href="https://github.com/arefbhrn">Aref Bahreini</a>
 */
class TimberLogTree(private val isDebugMode: Boolean) : Timber.Tree() {

    companion object {
        private const val DEFAULT_TAG = "TimberLogTree"
    }

    @SuppressLint("LogNotTimber")
    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
        var mTag = tag
        if (!isDebugMode) {
            return
        }
        mTag = mTag ?: DEFAULT_TAG
        when (priority) {
            Log.VERBOSE -> Log.v(mTag, message)
            Log.DEBUG -> Log.d(mTag, message)
            Log.INFO -> Log.i(mTag, message)
            Log.WARN -> Log.w(mTag, message)
            Log.ERROR -> Log.e(mTag, message)
        }
    }

    override fun formatMessage(message: String, args: Array<out Any?>): String {
        return String.format(Locale.ENGLISH, message, *args)
    }
}
